Brokerages/Binance/BinanceBrokerage.Utility.cs

```diff
@@ -0,0 +1,177 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using QuantConnect.Orders;
+using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace QuantConnect.Brokerages.Binance
+{
+    /// <summary>
+    /// Binance utility methods
+    /// </summary>
+    public partial class BinanceBrokerage
+    {
+        /// <summary>
+        /// Timestamp in milliseconds
+        /// </summary>
+        /// <returns></returns>
+        private long GetNonce()
+        {
+            return (long)(Time.TimeStamp() * 1000);
+        }
+
+        /// <summary>
+        /// Creates a signature for signed endpoints
+        /// </summary>
+        /// <param name="payload">the body of the request</param>
+        /// <returns>a token representing the request params</returns>
+        private string AuthenticationToken(string payload)
+        {
+            using (HMACSHA256 hmac = new HMACSHA256(Encoding.UTF8.GetBytes(ApiSecret)))
+            {
+                return ByteArrayToString(hmac.ComputeHash(Encoding.UTF8.GetBytes(payload)));
+            }
+        }
+
+        /// <summary>
+        /// If an IP address exceeds a certain number of requests per minute
+        /// HTTP 429 return code is used when breaking a request rate limit.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        private IRestResponse ExecuteRestRequest(IRestRequest request)
+        {
+            const int maxAttempts = 10;
+            var attempts = 0;
+            IRestResponse response;
+
+            do
+            {
+                if (!_restRateLimiter.WaitToProceed(TimeSpan.Zero))
```

Comment:
If this is the desired retry request behavior, it might make sense to extract into it's own type so other types can benefit from it.